### ISSUE TITLE
Refactor functions to utilize a debug module

### ISSUE THREAD

**Issue by nfahlgren** [danforthcenter/plantcv#713]
**Is your feature request related to a problem? Please describe.**
Pull request #712 creates a new private function `_debug` to replace the existing debug image display logic in most PlantCV functions. The new function has a single test function with two parameters, "print" and "plot". By utilizing the `_debug` function in existing PlantCV functions, a logical if/elif block can be removed from each and replaced with a single call to `_debug`. Tests for those functions can also be refactored to drop tests with "print" and "plot" enabled, which will reduce testing run time and complexity.

**Describe the solution you'd like**
PR #712, when merged will add the `_debug` private function without interfering with existing functions. It also refactors `pcv.readimage`, `pcv.analyze_object`, and `pcv.analyze_color` for testing purposes. I propose that, once #712 is merged, we open a new pull request per PlantCV function as time allows to update the rest of the code base over time. I recognize that is a lot of pull requests, but it will be quick to review and merge them, versus trying to update hundreds of files and tests in a single PR. If you are working on an existing function for other reasons, feel free to include these updates in that PR though.

For a given function, the general framework involves updating the function by removing imports to `print_image` and `plot_image` and adding an import for `_debug`. Then removing the logic around any debugging visuals and passing them, along with a filename to `_debug`. So for an existing pattern like:

```python
import os
from plantcv.plantcv import params
from plantcv.plantcv import print_image
from plantcv.plantcv import plot_image

# function code here

if params.debug is not None:
    # make a debug image
    if params.debug == "print":
        print_image(debug_img, os.path.join(params.debug_outdir, str(params.device) + "_function.png"))
    elif params.debug == "plot":
        plot_image(debug_img)
```

It can be simplified to:

```python
import os
from plantcv.plantcv import params
from plantcv.plantcv._debug import _debug

# function code here
# make a debug image
_debug(visual=debug_img, filename=os.path.join(params.debug_outdir, str(params.device) + "_function.png"))
```

Likewise, an existing test pattern like this:

```python
def test_plantcv_function():
    # Test cache directory
    cache_dir = os.path.join(TEST_TMPDIR, "test_plantcv_function")
    os.mkdir(cache_dir)
    pcv.params.debug_outdir = cache_dir
    # Test with debug = "print"
    pcv.params.debug = "print"
    _ = pcv.function()
    # Test with debug = "plot"
    pcv.params.debug = "plot"
    _ = pcv.function()
    # Test with debug = None
    pcv.params.debug = None
    output = pcv.function()
    assert output == expected
```

Can be simplified to:

```python
def test_plantcv_function():
    # Test with debug = None
    pcv.params.debug = None
    output = pcv.function()
    assert output == expected
```

# Function checklist for PRs

- [x] `pcv.acute_vertex` (@dschneiderch #807)
- [x] `pcv.analyze_bound_horizontal` (@dschneiderch #807)
- [x] `pcv.analyze_bound_vertical` (@dschneiderch #807)
- [x] `pcv.analyze_color` (@nfahlgren #712)
- [x] `pcv.analyze_nir_intensity` (@DannieSheng #726)
- [x] `pcv.analyze_object` (@nfahlgren #712)
- [x] `pcv.analyze_thermal_values` (@DannieSheng #726)
- [x] `pcv.apply_mask` (@JorgeGtz #817)
- [x] `pcv.auto_crop` (@dschneiderch #807)
- [x] `pcv.background_subtraction` (@dschneiderch #807)
- [x] `pcv.canny_edge_detect` (@dschneiderch #807)
- [x] `pcv.closing` (@dschneiderch #807)
- [x] `pcv.cluster_contour_splitimg` (@GrantKonkel  #725)
- [x] `pcv.cluster_contours` (@dschneiderch #807)
- [x] `pcv.crop` (@dschneiderch #807)
- [x] `pcv.crop_position_mask` (@dschneiderch #807)
- [x] `pcv.dilate` (@dschneiderch #807)
- [x] `pcv.distance_transform` (@dschneiderch #807)
- [x] `pcv.erode` (@dschneiderch #807)
- [x] `pcv.fill` (@jwheeler5 #794 )
- [x] `pcv.fill_holes` (@dschneiderch #807)
- [x] `pcv.find_objects` (@dschneiderch #807)
- [x] `pcv.flip` (@dschneiderch #807)
- [x] `pcv.gaussian_blur` (@dschneiderch #807)
- [x] `pcv.hist_equalization` (@dschneiderch #807)
- [x] `pcv.image_add` (@dschneiderch #807)
- [x] `pcv.image_subtract` (@dschneiderch #807)
- [x] `pcv.invert` (@dschneiderch #807)
- [x] `pcv.laplace_filter` (@dschneiderch #807)
- [x] `pcv.logical_and` (@dschneiderch #807)
- [x] `pcv.logical_or` (@dschneiderch #807)
- [x] `pcv.logical_xor` (@dschneiderch #807)
- [x] `pcv.median_blur` (@dschneiderch #807)
- [x] `pcv.naive_bayes_classifier` (@dschneiderch #807)
- [x] `pcv.object_composition` (@dschneiderch #807)
- [x] `pcv.opening` (@dschneiderch #807)
- [ ] `pcv.output_mask` (@nfahlgren #848)
- [x] `pcv.readbayer` (@dschneiderch #807)
- [x] `pcv.readimage` (@nfahlgren #712)
- [x] `pcv.rectangle_mask` (@dschneiderch #807)
- [x] `pcv.report_size_marker_area` (@dschneiderch #807)
- [x] `pcv.rgb2gray` (@adrianethompson #776)
- [x] `pcv.rgb2gray_hsv` (@dschneiderch #807)
- [x] `pcv.rgb2gray_lab` (@dschneiderch #807)
- [x] `pcv.roi_objects` (@dschneiderch #807)
- [x] `pcv.scale_features` (@dschneiderch #807)
- [x] `pcv.scharr_filter` (@dschneiderch #807)
- [x] `pcv.shift_img` (@dschneiderch #807)
- [x] `pcv.sobel_filter` (@dschneiderch #807)
- [x] `pcv.spatial_clustering` (@dschneiderch #807)
- [x] `pcv.stdev_filter` (@dschneiderch #807)
- [x] `pcv.watershed_segmentation` (@dschneiderch #807)
- [x] `pcv.white_balance` (@dschneiderch #807)
- [x] `pcv.x_axis_pseudolandmarks` (@dschneiderch #807)
- [x] `pcv.y_axis_pseudolandmarks` (@dschneiderch #807)
- [x] `pcv.hyperspectral.analyze_index` (@DannieSheng #726 )
- [x] `pcv.hyperspectral.analyze_spectral` (@DannieSheng #726 )
- [x] `pcv.hyperspectral.calibrate` (@nfahlgren #845)
- [ ] `pcv.hyperspectral.extract_wavelength` (@nfahlgren #848)
- [x] `pcv.hyperspectral.read_data`(@DannieSheng #739 )
- [ ] `pcv.morphology.analyze_stem` (@nfahlgren #848)
- [ ] `pcv.morphology.check_cycles` (@nfahlgren #848)
- [x] `pcv.morphology.fill_segments`(@JorgeGtz #741 )
- [ ] `pcv.morphology.find_branch_pts` (@nfahlgren #848)
- [ ] `pcv.morphology.find_tips` (@nfahlgren #848)
- [ ] `pcv.morphology.prune` (@nfahlgren #848)
- [ ] `pcv.morphology.segment_angle` (@nfahlgren #848)
- [ ] `pcv.morphology.segment_combine` (@nfahlgren #848)
- [ ] `pcv.morphology.segment_curvature` (@nfahlgren #848)
- [ ] `pcv.morphology.segment_euclidean_length` (@nfahlgren #848)
- [ ] `pcv.morphology.segment_id` (@nfahlgren #848)
- [ ] `pcv.morphology.segment_insertion_angle` (@nfahlgren #848)
- [ ] `pcv.morphology.segment_path_length` (@nfahlgren #848)
- [ ] `pcv.morphology.segment_skeleton` (@nfahlgren #848)
- [ ] `pcv.morphology.segment_sort` (@nfahlgren #848)
- [ ] `pcv.morphology.segment_tangent_angle` (@nfahlgren #848)
- [ ] `pcv.morphology.skeletonize` (@nfahlgren #848)
- [x] `pcv.photosynthesis.analyze_fvfm` (@HaleySchuhl #750)
- [x] `pcv.photosynthesis.read_cropreporter` (@HaleySchuhl #750)
- [x] `pcv.roi.roi2mask` (@dschneiderch #807)
- [x] `pcv.roi.from_binary_image` (@dschneiderch #807)
- [x] `pcv.roi.rectangle` (@dschneiderch #807)
- [x] `pcv.roi.circle` (@dschneiderch #807)
- [x] `pcv.roi.ellipse` (@dschneiderch #807)
- [x] `pcv.roi._draw_roi` (@dschneiderch #807)
- [x] `pcv.roi.multi` (@dschneiderch #807)
- [x] `pcv.roi.custom` (@dschneiderch #807)
- [x] `pcv.spectral_index._package_index` (@nfahlgren #848)
- [x] `pcv.threshold.custom_range` (@DannieSheng #569 )
- [x] `pcv.threshold._call_threshold` (@DannieSheng #569 )
- [x] `pcv.threshold._call_adaptive_threshold` (@DannieSheng #569)
- [x] `pcv.threshold.saturation` (@DannieSheng #569)
- [ ] `pcv.transform.apply_transformation_matrix` (@nfahlgren #848)
- [ ] `pcv.transform.create_color_card_mask` (@nfahlgren #848)
- [ ] `pcv.transform.quick_color_check` (@nfahlgren #848)
- [ ] `pcv.transform.nonuniform_illumination` (@nfahlgren #848)
- [ ] `pcv.transform.rescale` (@nfahlgren #848)
- [ ] `pcv.transform.resize` (@nfahlgren #848)
- [ ] `pcv.transform.resize_factor` (@nfahlgren #848)
- [ ] `pcv.transform.rotate` (@nfahlgren #848)
- [x] `pcv.transform.warp` (@DannieSheng #720)
- [x] `pcv.visualize.auto_threshold_methods` (@HaleySchuhl)
- [ ] `pcv.visualize.clustered_contours` (@nfahlgren #848)
- [x] `pcv.visualize.colorize_masks` (@HaleySchuhl)
- [ ] `pcv.visualize.colorspaces` (@nfahlgren #848)
- [x] `pcv.visualize.histogram` (@DannieSheng #726)
- [x] `pcv.visualize.overlay_two_imgs` (@DannieSheng #734 )

**Comment by DannieSheng** [comment]
Addressed in #720
